src/storage/GraphStorageLocalServer.cpp

```diff
@@ -0,0 +1,155 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ */
+#include "GraphStorageLocalServer.h"
+
+#include <unistd.h>
+
+#include "common/base/Base.h"
+
+namespace nebula::storage {
+
+std::mutex mutex_;
+std::shared_ptr<GraphStorageLocalServer> instance_ = nullptr;
+
+void GraphStorageLocalServer::setPort(uint16_t port) { UNUSED(port); }
+
+void GraphStorageLocalServer::setIdleTimeout(std::chrono::milliseconds timeout) { UNUSED(timeout); }
+
+void GraphStorageLocalServer::setIOThreadPool(
+    std::shared_ptr<folly::IOThreadPoolExecutor> ioThreadPool) {
+  UNUSED(ioThreadPool);
+}
+
+void GraphStorageLocalServer::setThreadManager(
+    std::shared_ptr<apache::thrift::concurrency::ThreadManager> threadManager) {
+  // lock?
+  threadManager_ = threadManager;
+}
+
+void GraphStorageLocalServer::setStopWorkersOnStopListening(bool stopWorkers) {
+  UNUSED(stopWorkers);
+}
+
+void GraphStorageLocalServer::setInterface(std::shared_ptr<GraphStorageServiceHandler> handler) {
+  handler_ = handler;
+}
+
+void GraphStorageLocalServer::setSSLConfig(std::shared_ptr<wangle::SSLContextConfig> context) {
+  UNUSED(context);
+}
+
+void GraphStorageLocalServer::serve() {
+  if (!stop_) {
+    return;  // error happend
+  }
+  stop_ = false;
+  while (true) {  // temp implment
+    sleep(1);
+    if (stop_) {
+      break;
+    }
+  }
+}
+
+folly::Future<::nebula::storage::cpp2::GetNeighborsResponse>
+GraphStorageLocalServer::future_getNeighbors(
+    const ::nebula::storage::cpp2::GetNeighborsRequest& p_req) {
+  return folly::via(threadManager_.get(),
```

Comment:
In normal deployment, we have another thread pool called `readerPool`, this is used for `read` query separately. (write could be quite slow and occupy many worker thread in threadManager_, so read query need another thread pool).

Perhaps we could just use the `threadManager_` in `GraphStorageLocalServer`, I'll think more about it. You could pay more attention when we do perf test later.

Reply:
Perhaps there will be another deadlock case. For example lots of `go 1 steps over xxx yield $$.tag.property.`, and all worker thread has been occupied in `GraphStorageLocalServer` when we do the `go`. Before we move on to get the dst's property, here comes another bunch of query, so I don't know whether we can continue to do the `yield $$.tag.property` part.